starcheck/check_obsid.py

```diff
@@ -0,0 +1,406 @@
+    # subtract off the offsets.
+    fid_ids = np.array(kw['fid_ids'])
+    if kw['detector'] == 'HRC-I':
+        fid_offset = 6
+    elif kw['detector'] == 'HRC-S':
+        fid_offset = 10
+    else:
+        fid_offset = 0
+    fid_ids -= fid_offset
+
+    args = dict(obsid=int(kw['obsid']),
+                att=Quaternion.normalize(kw['att']),
+                date=kw['date'],
+                n_acq=kw['n_acq'],
+                n_guide=kw['n_guide'],
+                man_angle=kw['man_angle'],
+                t_ccd_acq=kw['t_ccd_acq'],
+                t_ccd_guide=kw['t_ccd_guide'],
+                dither_acq=ACABox(kw['dither_acq']),
+                dither_guide=ACABox(kw['dither_guide']),
+                include_ids_acq=kw['include_ids_acq'],
+                include_halfws_acq=kw['include_halfws_acq'],
+                detector=kw['detector'], sim_offset=kw['sim_offset'],
+                include_ids_guide=kw['include_ids_guide'],
+                include_ids_fid=list(fid_ids),
+                n_fid=len(kw['fid_ids']), focus_offset=0)
+    if 'monitors' in kw:
+        args['monitors'] = kw['monitors']
+    aca = get_aca_catalog(**args)
+    #import pickle
+    #pickle.dump({int(kw['obsid']): aca}, open(f"{kw['date']}.pkl", 'wb'))
+    return aca
+
+
+def proseco_probs(aca, include_ids_acq):
+    acq_cat = aca.acqs
+
+    # Assign the proseco probabilities back into an array.
+    p_acqs = [float(acq_cat['p_acq'][acq_cat['id'] == acq_id][0]) for acq_id in include_ids_acq]
+
+    return p_acqs, float(-np.log10(acq_cat.calc_p_safe())), float(np.sum(p_acqs))
+
+
+def run_sparkles(aca):
+    acar = aca.get_review_table()
+    acar.run_aca_review()
+    return {'warn': [w['text'] for w in acar.messages == 'critical'],
+            'orange_warn': [w['text'] for w in acar.messages == 'caution'],
+            'yellow_warn': [w['text'] for w in acar.messages == 'warning'],
+            'fyi': [w['text'] for w in acar.messages == 'info']}
+
+
+def _mag_for_p_acq(p_acq, date, t_ccd):
+    """
+    Call mag_for_p_acq, but cast p_acq and t_ccd as floats (may or may not be needed) and
+    convert date from a bytestring (from the Perl interface).
+    """
+    eff_t_ccd = get_effective_t_ccd(t_ccd)
+    return mag_for_p_acq(float(p_acq), date.decode(), float(eff_t_ccd))
+
+
+def manual_stars(obsid, proseco_file):
+    """
+    Get the manual include/excludes from the products pickle to display.
+    """
+    pth = Path(de_bytestr(proseco_file))
+    open_func = open if pth.suffix == '.pkl' else gzip.open
+    manual_entries = []
+    with open_func(pth, 'rb') as fh:
+        acas_dict = pickle.load(fh)
+        aca = acas_dict[obsid]
+        call_args = aca.call_args.copy()
+        for key in call_args:
+            if key.startswith('include') or key.startswith('exclude'):
+                manual_entries.append(f'has {key} {call_args[key]}\n')
+    return manual_entries
+
+
+def compare_cats(cat1, cat2, type=None):
+
```

Comment:
This was written before the proseco catalog diff... not sure if it should stay or go at this point.